### PR TITLE
Support on-disk instance segmentations in SDK

### PR DESCRIPTION
**Pull request by brimoor**
```py
import fiftyone as fo
import fiftyone.zoo as foz
import fiftyone.utils.labels as foul

dataset = foz.load_zoo_dataset(
    "coco-2017",
    split="validation",
    label_types="segmentations",
    classes=["cat", "dog"],
    label_field="instances",
    max_samples=10,
    only_matching=True,
)

foul.export_segmentations(
    dataset,
    "instances",
    "/tmp/instances/masks",
)

assert dataset.count("instances.detections.mask") == 0
assert dataset.count("instances.detections.mask_path") > 0

dataset.export(
    export_dir="/tmp/instances/fod",
    dataset_type=fo.types.FiftyOneDataset,
    # export_media=False,
)

dataset2 = fo.Dataset.from_dir(
    dataset_dir="/tmp/instances/fod",
    dataset_type=fo.types.FiftyOneDataset,
)

assert dataset2.count("instances.detections.mask") == 0
assert dataset2.count("instances.detections.mask_path") > 0

foul.import_segmentations(dataset2, "instances")

assert dataset2.count("instances.detections.mask") > 0
assert dataset2.count("instances.detections.mask_path") == 0
```

## Summary by CodeRabbit

- **New Features**
	- Enhanced documentation for FiftyOne datasets, including new sections and clearer examples.
	- Improved functionality for dataset exporters and importers, supporting additional label types.
	- Added new test methods for validating instance segmentation functionality.

- **Bug Fixes**
	- Clarified requirements for segmentation processes and improved error handling in related methods.

- **Documentation**
	- Expanded and refined sections on instance segmentation, dataset deletion, and dynamic attributes for better user understanding.

- **Tests**
	- Introduced new unit tests for instance segmentation to ensure robust functionality.

### CODE REVIEW

File: fiftyone/core/labels.py [hunk 1]
```diff
@@ -409,7 +409,8 @@ class Detection(_HasAttributesDict, _HasID, _HasMedia, Label):
             its bounding box, which should be a 2D binary or 0/1 integer numpy
             array
         mask_path (None):  the absolute path to the instance segmentation image
-            on disk
+            on disk, which should be a single-channel PNG image where any
+            non-zero values represent the instance's extent
```
Comment (sashankaryal) [inline fiftyone/core/labels.py:413]: the app technically doesn't mind multi-channel, too, but it makes sense that we write an imperative statement about masks being single-channel for clarity.

if the app runs into multi-channel pngs for masks, it uses just the first channel

**Review by sashankaryal — DISMISSED** [review]
LGTM! Thanks for the follow-up work - I missed a lot. 🥲

**Review by benjaminpkane — DISMISSED** [review]
LGTM

File: fiftyone/core/labels.py (hunk 1 above)
Comment (brimoor) [reply fiftyone/core/labels.py:413]: Ah I see. I mostly copied this verbatim from the user guide documentation you had added:

> We recommend storing masks as single-channel PNG images, where a pixel value of 0 indicates the background (rendered as transparent in the App), and any other value indicates the object.

But I was not certain whether single-channel was indeed a recommendation or a hard requirement, so the version here came out sounding more imperative just to be safe.

Comment (brimoor) [reply fiftyone/core/labels.py:413]: Feel free to clarify both instances of this documentation if you want!

**Review by sashankaryal — APPROVED** [review]
let's go! 🚀

**Pull request merged**